ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +216,36 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)]
+      {:content        (delay (.getInputStream conn))
```

Comment:
We only use :content, :content-length, and :last-modified if :directory? is false. It's probably safe to assume that the first two are easy to determine for directory entries, but is is safe to do so for :last-modified?